rosetta-source/src/main/rosetta/product-collateral-func.rosetta

```diff
@@ -1,6 +1,13 @@
 namespace cdm.product.collateral
 version "${project.version}"
 
+import cdm.base.math.*
+import cdm.base.datetime.*
+import cdm.base.datetime.daycount.*
+import cdm.base.staticdata.party.*
+import cdm.legaldocumentation.common.CreditSupportAgreementMarginTypeEnum
```

Comment:
Rename `CreditSupportAgreementMarginTypeEnum` as simply `MarginTypeEnum` and move to the `product.collateral` namespace, so that you don't have to import from the `legaldocumentation` namespace. `legaldocumentation` is meant to sit downstream from `product` in the CDM namespace structure, so this would create a pseudo "circular dependency".

Reply:
Will the CreditSupportAgreementMarginTypeEnum remain unchanged in the ISDA CDM within the legal agreement structure that supports the CSA as this should be left as CreditSupportAgreementMarginTypeEnum in the foundations version